plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/XtextAntlrGeneratorFragment2.xtend

```diff
@@ -44,26 +44,27 @@ import org.eclipse.xtext.xtext.generator.model.FileAccessFactory
 import org.eclipse.xtext.xtext.generator.model.GuiceModuleAccess
 import org.eclipse.xtext.xtext.generator.model.IXtextGeneratorFileSystemAccess
 import org.eclipse.xtext.xtext.generator.model.JavaFileAccess
+import org.eclipse.xtext.xtext.generator.model.TypeReference
+import org.eclipse.xtext.xtext.generator.util.BooleanGeneratorOption
+import org.eclipse.xtext.xtext.generator.util.SyntheticTerminalDetector
 
 import static extension org.eclipse.xtext.GrammarUtil.*
 import static extension org.eclipse.xtext.xtext.generator.model.TypeReference.*
 import static extension org.eclipse.xtext.xtext.generator.parser.antlr.AntlrGrammarGenUtil.*
-import org.eclipse.xtext.xtext.generator.util.SyntheticTerminalDetector
-import org.eclipse.xtext.xtext.generator.model.TypeReference
 
 class XtextAntlrGeneratorFragment2 extends AbstractAntlrGeneratorFragment2 {
-	@Accessors
+	
+	@Accessors(PUBLIC_SETTER)
 	boolean debugGrammar
 	
-	@Accessors
-	Boolean combinedGrammar = null
+	val combinedGrammar = new BooleanGeneratorOption
 	
-	@Accessors
+	@Accessors(PUBLIC_SETTER)
 	boolean removeBacktrackingGuards
```

Comment:
Then we should add a protected getter. I removed public getters where possible in order to reduce the exposed api.